File: hw/nase/u3/hw23AbstractFactory/AbstractFactory.JavaScript/model/GUIFactory.js
import fs from 'fs';
import path from 'path';

export class GUIFactory {
    static async getFactory() {
        const sys = GUIFactory.readFromFile("OS_TYPE");
        if (sys === 0) {
            const { WinFactory } = await import('./WinFactory.js');
            return new WinFactory();
        } else {
            const { LinuxFactory } = await import('./LinuxFactory.js');
            return new LinuxFactory();
        }
    }
 
    static readFromFile(key) {
        try {
            const filePath = path.resolve('config.properties');
            if (!fs.existsSync(filePath)) return 0;
            
            const data = fs.readFileSync(filePath, 'utf-8');
            const lines = data.split('\n');
            for (let line of lines) {
                line = line.trim();
                if (line && !line.startsWith('#')) {
                    const [k, v] = line.split('=');
                    if (k.trim() === key) {
                        return parseInt(v.trim(), 10);
                    }
                }
            }
            return 0;
        } catch (error) {
            return 0;
        }
    }

    createButton() {
        throw new Error("Method 'createButton()' must be implemented.");
    }

    createMenu() {
        throw new Error("Method 'createMenu()' must be implemented.");
    }
}

File: hw/nase/u3/hw23AbstractFactory/AbstractFactory.python/model/GUIFactory.py
from abc import ABC, abstractmethod

class GUIFactory(ABC):
    
    @staticmethod
    def get_factory():
        sys = GUIFactory.read_from_file("OS_TYPE")
        if sys == 0:
            import model.WinFactory as wf  
            return wf.WinFactory()        
        else:
            import model.LinuxFactory as lf  
            return lf.LinuxFactory()         

    @staticmethod
    def read_from_file(key):
        try:
            with open("config.properties", "r") as f:
                for line in f:
                    if line.strip() and not line.startswith("#"):
                        k, v = line.split("=")
                        if k.strip() == key:
                            return int(v.strip())
            return 0
        except (FileNotFoundError, ValueError):
            return 0

    @abstractmethod
    def create_button(self):
        pass

    @abstractmethod
    def create_menu(self):
        pass